### ISSUE TITLE
SEGV when loading function multiversioning enabled binary from dlopen

### ISSUE THREAD

**Issue by kazuki** [jubatus/jubatus_core#281]
linked executable binary (jubaclassifier, ...) is works fine. But dlopen is failed with SEGV.
This issue affects my project. https://github.com/kazuki/embedded-jubatus-python/

I tested with gcc-5.3.0/glibc-2.22 (gentoo) only.
other gcc/glibc version, this issue may not reproduce.

(update) may be old-gcc+ld.bfd issue because ld.gold not reproduced.
- Gentoo(gcc-5.3.0/binutils-2.25): ld.bfd (2.25): NG
- Gentoo(gcc-5.3.0/binutils-9999): ld.bfd (trunk, 2.26.x): NG

(update) below environments works fine !
- Ubuntu 16.04 (quickhack #269): gcc-5.3.1-14ubuntu2, glibc-2.23-0ubuntu3, binutils-2.26(ld.bfd)
- Gentoo: gcc-5.3.0/glibc-2.22/binutils-2.25(ld.gold)
- Gentoo: gcc-5.4.0/glibc-2.22/binutils-2.25.1(ld.bfd / ld.gold)

**Steps to Reproduce**
1. builds jubatus_core with function multiversioning (gcc-5.3)
2. dlopen libjubatus_core.so

```
$ python -c "import ctypes; ctypes.cdll.LoadLibrary('libjubatus_core.so')"
```

**Actual Results**

segmentation fault

**Expected Results**

dlopen success

**Backtrace**

```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000105976 in ?? ()
(gdb) bt
#0  0x0000000000105976 in ?? ()
#1  0x00007f7d2f7bbdd9 in jubatus::core::storage::detail::(anonymous namespace)::calc_hamming_distance_impl () at ../jubatus/core/storage/bit_vector.cpp:110
#2  0x00007f7d3293de40 in _dl_relocate_object () from /lib64/ld-linux-x86-64.so.2
#3  0x00007f7d329457a9 in dl_open_worker () from /lib64/ld-linux-x86-64.so.2
#4  0x00007f7d329410d4 in _dl_catch_error () from /lib64/ld-linux-x86-64.so.2
#5  0x00007f7d32944fc1 in _dl_open () from /lib64/ld-linux-x86-64.so.2
#6  0x00007f7d32512fd9 in dlopen_doit () from /lib64/libdl.so.2
#7  0x00007f7d329410d4 in _dl_catch_error () from /lib64/ld-linux-x86-64.so.2
#8  0x00007f7d325135d1 in _dlerror_run () from /lib64/libdl.so.2
#9  0x00007f7d32513071 in dlopen () from /lib64/libdl.so.2
```

**Comment by yukimori** [comment]
Thank you for your information.
This issue doesn't occur with ubuntu12.04 and ubuntu14.04.
We will try to find the cause of this issue, but it will take a time since we know little about ld.

**Comment by kazuki** [comment]
Fixed(workaround) via #347
